// Copyright (C) 2017, Yeolar

syntax = "proto3";
option cc_generic_services = true;

package rdd.pbrpc;

message Query {
  string traceid = 1;
  string query = 2;
  string forward = 3;
}

enum ResultCode {
  OK = 0;

  // WARNING      <1000

  // ERROR        >1000
  E_SOURCE__UNTRUSTED = 1001;
  E_BACKEND_FAILURE = 1002;
}

message Result {
  string traceid = 1;
  ResultCode code = 2;
}

service ProxyService {
  rpc run(Query) returns(Result);
}
